### PR TITLE
Feature/get lead times

### PR DESCRIPTION
**Pull request by krisquigley**

### CODE REVIEW

File: lib/actions/getVendorsByReference.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+const getVendorReferences  = require('../getVendorsByReference/getVendorReferences.js');
+const leadTimes  = require('../getVendorsByReference/filterLeadTimes.js');
+const messages       = elasticio.messages;
+
+exports.process = processAction;
+
+function processAction(message, config) {
+  let self = this;
+
+  // submits the HTTP request to the VMS to acknowledge the order
```
Comment (hubertpompecki) [inline lib/actions/getVendorsByReference.js]: I think this comment is misleading

File: lib/actions/getVendorsByReference.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+"use strict";
+const request        = require('request');
+const elasticio      = require('elasticio-node');
```
Comment (hubertpompecki) [inline lib/actions/getVendorsByReference.js]: This is not needed.

File: lib/actions/getVendorsByReference.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+const requestOptions = require('../requestOptions.js');
+const handleResponse = require('../handleResponse.js');
+const getVendorReferences  = require('../getVendorsByReference/getVendorReferences.js');
+const leadTimes  = require('../getVendorsByReference/filterLeadTimes.js');
+const messages       = elasticio.messages;
```
Comment (hubertpompecki) [inline lib/actions/getVendorsByReference.js]: This is not needed.

File: spec/getVendorsByReferenceSpec.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+const getVendorsByReference = require('../lib/actions/getVendorsByReference.js');
+const orderWebhook = require('./fixtures/orderWebhook.json');
+const getVendorsByReferenceResponse = require('./fixtures/getVendorsResponse.json');
+const elasticio = require('elasticio-node');
```
Comment (hubertpompecki) [inline spec/getVendorsByReferenceSpec.js]: This is not needed.

File: spec/getVendorsByReferenceSpec.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+const filterLeadTimes  = require('../lib/getVendorsByReference/filterLeadTimes.js');
+const messages = elasticio.messages;
```
Comment (hubertpompecki) [inline spec/getVendorsByReferenceSpec.js]: This is not needed.

File: spec/getVendorsByReferenceSpec.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+describe("Retrieve vendors lead times", () => {
+
+  beforeEach(() => {
+    this.message = {
+      body: orderWebhook
+    }
+
+    this.config = {
+      host:   "https://vendors-staging.herokuapp.com",
+      apiKey: "c91966a95dfdf3f6a446dedc189f49c75827e20f84b30c24d22749810695d86c"
```
Comment (hubertpompecki) [inline spec/getVendorsByReferenceSpec.js]: We might want to replace these credentials with some dummy data given that we are mocking everything out with nock.

File: lib/getVendorsByReference/filterLeadTimes.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    let data = {};
+
+    body.data.forEach((vendor) => {
+      data[vendor.attributes.name] = vendor.attributes.leadtime
+    })
+
+    return data;
+  },
+
+  returnLeadTimes: (body) => {
```
Comment (hubertpompecki) [inline lib/getVendorsByReference/filterLeadTimes.js:15]: Is this function used anywhere?

**Review by hubertpompecki — APPROVED** [review]
:shipit:

File: lib/getVendorsByReference/filterLeadTimes.js (hunk 7 above)
Comment (krisquigley) [reply lib/getVendorsByReference/filterLeadTimes.js:15]: It's used in the action

**Comment by krisquigley** [comment]
Updated based on comments and merged

**Pull request merged**